TypeCobol/Compiler/Types/RenamesType.cs

```diff
@@ -0,0 +1,122 @@
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Symbols;
+
+namespace TypeCobol.Compiler.Types
+{
+    /// <summary>
+    /// This is a renamed type, a specialization of a RecordType.
+    /// 
+    /// RENAMES symbol used to regrouping elementary items in a group item.
+    /// RENAMES will create a logical group from the group of elementary items.
+    /// 
+    /// Syntax:
+    ///     66 data-name-1 RENAMES data-name-2
+    ///         [THROUGH/THRU data-name-3]
+    ///  
+    /// data-name-1 is the alternative name for the logcal group.
+    /// data-name-2 is the starting elementary item and data-name-3 is the ending elementary item
+    ///             in the baic group.
+    ///  When data-name-3 is not specified, the data-name-2 must be a group item
+    ///  and all the elementary items under this RENAMED to data-name-1.
+    ///  
+    /// Rules:
+    ///     1. RENAMES entries must be in a sequential order
+    ///     2. 66 level numbers didn't have a PIC clause
+    ///     3. RENAMES clause mut be coded at the end of the group
+    ///     4. Level 66 entry cannot rename level-01, level-77, level-88, or another level 66-entry.
+    ///     5. Elementary items with OCCURS clause should not be RENAMED.
+    ///     
+    /// Exemples:
+    ///     01 A.
+    ///         05 ITEM1 PIC X(5).
+    ///         05 ITEM2 PIC X(5).
+    ///         05 ITEM3 PIC X(5).
+    ///         05 ITEM4 PIC X(5).
+    ///         05 ITEM5 PIC X(5).
+    ///         05 ITEM6 PIC X(5).
+    ///         05 ITEM7 PIC X(5).
+    ///         05 ITEM8 PIC X(5).
+    ///         05 ITEM9 PIC X(5).
+    ///         05 ITEM10 PIC X(5).
+    ///     66 B RENAMES ITEM1 THRU ITEM6.
+    /// </summary>
+    /// </summary>
+    public class RenamesType : GroupType
+    {
+        /// <summary>
+        /// Constructor for the Syntax:
+        ///     66 data-name-1 RENAMES data-name-2
+        /// 
+        /// 
+        /// </summary>
+        /// <param name="owner">Owner of the scope</param>
+        /// <param name="containerType"></param>
+        /// <param name="data_name_2"></param>
+        /// <param name="data_name_3"></param>
+        public RenamesType(Symbol owner, GroupType containerType, VariableSymbol data_name_2) : this(owner, containerType, data_name_2, data_name_2)
+        {
+        }
+
+        /// <summary>
+        /// Constructor for the Syntax:
+        ///     66 data-name-1 RENAMES data-name-2
+        ///         [THROUGH/THRU] data-name-3
+        /// 
+        /// 
+        /// </summary>
+        /// <param name="owner">Owner of the scope</param>
+        /// <param name="containerType"></param>
+        /// <param name="data_name_2"></param>
+        /// <param name="data_name_3"></param>
+        public RenamesType(Symbol owner, GroupType containerType, VariableSymbol data_name_2, VariableSymbol data_name_3) : base(owner)
+        {
+            ContainerType = containerType;
+            DataName1 = data_name_2;
```

Comment:
Because of how a Renames variable can be declared:
Constructor for the Syntax:
     66 data-name-1 RENAMES data-name-2
         [THROUGH/THRU] data-name-3

Reply:
Yes but we should use more explicit names like 'From' and 'To' or 'StartVariable' and 'EndVariable' because those properties refer to boundaries of the rename.